core/arch/arm/kernel/thread.c

```diff
@@ -771,22 +771,16 @@ static void init_thread_stacks(void)
 		tee_pager_add_pages(tee_mm_get_smem(mm), tee_mm_get_size(mm),
 				    true);
 
-		/* Realloc both protection vmem and stack vmem separately */
-		sp = tee_mm_get_smem(mm);
-		tee_mm_free(mm);
-		mm = tee_mm_alloc2(&tee_mm_vcore, sp, SMALL_PAGE_SIZE);
-		TEE_ASSERT(mm);
-		mm = tee_mm_alloc2(&tee_mm_vcore, sp + SMALL_PAGE_SIZE,
-						  STACK_THREAD_SIZE);
-		TEE_ASSERT(mm);
+		/* Add the area to the pager */
+		tee_pager_add_core_area(tee_mm_get_smem(mm) + SMALL_PAGE_SIZE,
```

Comment:
Why  do we add SMALL_PAGE_SIZE? Is it because the first page already has been populated in ` core/arch/arm/kernel/generic_boot.c : init_runtime` for this area?